File: CTCI_1_String_Array/ctci_1_1.py
def isUnique(s):
	s_list = list(s)
	s_set = set(s_list)
	len_list = len(s_list)
	len_set = len(s_set)
	return len_list == len_set

def main():
	s1 = 'i have a dream'
	s2 = 'qwertyuiop'
	print isUnique(s1), isUnique(s2)

if __name__ == '__main__':
	main()

File: CTCI_1_String_Array/ctci_1_1.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

bool isUnique(string s)
{
	vector<bool> table(256, false);
	for(int i = 0; i < s.size(); i++)
	{
		int pos = (int)s[i];
		if(table[pos] == true)
			return false;
		table[pos] = true;
	}
	return true;
}

int main(int argc, char const *argv[])
{
	string s1 = "i have a dream";
	string s2 = "qwertyuiop";
	cout<<isUnique(s1)<<" "<<isUnique(s2)<<endl;
	return 0;
}